framework/src/timesteppers/AB2PredictorCorrector.C

```diff
@@ -85,8 +85,7 @@ AB2PredictorCorrector::step()
   NonlinearSystemBase & nl = _fe_problem.getNonlinearSystemBase();
   AuxiliarySystem & aux = _fe_problem.getAuxiliarySystem();
 
-  _fe_problem.solve();
-  _converged = _fe_problem.converged();
+  _converged = _fe_problem.newSolve();
```

Comment:
This is a little odd. We have newSolve() that returns a converged status but also a converged() call in NonlinearSystemBase()? Are we sure that's where we want the converge() method or should we just leave it at the FEProblem(Base) level?

Reply:
This is the complicated part. What I have done in this PR is that when `FEProblemBase::solve` is done, it returns true/false to indicate if the solve successful or not. It also set the status of the solver `_picard_status` like how the solve is converged or diverged, which can be inquired by `FEProblemBase::checkConvergence`. In the future, users may want to check more status of the solve, like convergence history, etc, so keeping `FEProblemBase::checkConvergence` makes sense. Returning true/false makes things handy although we can definitely use `FEProblemBase::checkConvergence` to inquire the convergence.

I guess leaving `NonlinearSystemBase::converged` which indicates if nonlinear solver is converged is fine, plus phase-field module and few apps are directly checking that.